src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -3273,12 +3274,18 @@ private async Task<(long Index, object Result)> SendToLeaderAsyncInternal(Transa
                 }
                 catch (Exception ex)
                 {
-                    if (Logger.IsInfoEnabled)
-                        Logger.Info($"Tried to send message to leader (reached: {reachedLeader.Value}), retrying", ex);
+                    /*
+                        if reachedLeader.Value is true and ex is ConcurrencyException meaning The leader was changed during the PutAsync
+                        (RachisMergedCommand try to insert to leaders log, but its leader is outdated (has old term)) ,So we will retry.
+                        reachedLeader.Value is false we'll retry also, because this is not the leader
+                    */
 
-                    if (reachedLeader.Value)
+                    if (reachedLeader.Value && ex is not ConcurrencyException)
```

Comment:
I think this is not good because the `ConcurrencyException` is too generic here and might cause an infinite loop.
I want you to add a test that will throw `RachisConcurrencyException` on `AddDatabaseCommand`.

Reply:
Done